lib/engine/game/g_18_uruguay/nationalization.rb

```diff
@@ -0,0 +1,286 @@
+              G18Uruguay::Step::DiscardTrain,
+              G18Uruguay::Step::PayoffLoans,
+              G18Uruguay::Step::NationalizeCorporation,
+              G18Uruguay::Step::RemoveTokens,
+              ], round_num: round_num)
+        end
+
+        def affected_shares(entity, corps)
+          affected = entity.shares.select { |s| s.corporation == corps.first }.sort_by(&:percent).reverse
+          unless corps.one?
+            affected.concat(entity.shares.select do |s|
+                              s.corporation == corps.last
+                            end.sort_by(&:percent).reverse)
+          end
+          affected
+        end
+
+        def find_president(holders, corps)
+          president_candidate = nil
+          candidate_sum = 0
+          holders.each do |holder|
+            entity_shares = affected_shares(holder, corps)
+            sum = entity_shares.sum(&:percent)
+            if sum > candidate_sum
+              president_candidate = holder
+              candidate_sum = sum
+            end
+          end
+          president_candidate
+        end
+
+        def transfer_share(share, new_owner)
+          corp = share.corporation
+          corp.share_holders[share.owner] -= share.percent
+          corp.share_holders[new_owner] += share.percent
+          share.owner.shares_by_corporation[corp].delete(share)
+          new_owner.shares_by_corporation[corp] << share
+          share.owner = new_owner
+        end
+
+        def transfer_pres_share(corporation, owner)
+          pres_share = corporation.presidents_share
+          transfer_share(pres_share, owner)
+          corporation.owner = owner
+        end
+
+        def acquire_shares_in_fce(corp_fce, merge_data)
+          new_president = find_president(merge_data[:holders], merge_data[:corps])
+          transfer_pres_share(corp_fce, new_president)
+
+          merge_data[:holders].each do |holder|
+            aquired = 0
+            aquired = 20 if holder == new_president
+            entity_shares = affected_shares(holder, merge_data[:corps])
+            total_percent = entity_shares.sum(&:percent)
+            aquire_percent = (total_percent / 20).to_i * 10
+            while aquired < aquire_percent
+              share = corp_fce.shares.first
+              aquired += share.percent
+              transfer_share(share, holder)
+            end
+            number = aquire_percent
+            @log << "#{holder.name} recives #{number}% in FCE in exchange to the nationalized shares"
+            odd_share = aquired * 2 != total_percent
+            next unless odd_share
+
+            price = corp_fce.share_price.price / 2
+            @bank.spend(price, holder)
+            @log << "#{holder.name} recives #{price} from halv share"
+          end
+          @log << "#{new_president.name} becomes new president of #{corp_fce.name}"
+        end
+
+        def compute_merger_share_price(corp_a, corp_b)
+          price = corp_a.share_price.price
```

Comment:
```suggestion
```

Reply:
No cannot do that. We need to handle that only one corp that merges to FCE

Reply:
Replaced with sum instead